cmake/modules/FindZephyr-sdk.cmake

```diff
@@ -103,10 +87,10 @@ if(NOT DEFINED ZEPHYR_TOOLCHAIN_VARIANT)
     #  Note: When CMake minimum version becomes >= 3.17, change this loop into:
     #    foreach(version config IN ZIP_LISTS Zephyr-sdk_CONSIDERED_VERSIONS Zephyr-sdk_CONSIDERED_CONFIGS)
     set(error_msg "The Zephyr SDK version you are using is not supported, please update your SDK.\n")
-    set(missing_version "You need SDK version ${TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION} or newer.")
```

Comment:
When I force 0.14.2 I don't see this, I get this output:
```
CMake Error at /tmp/aa/zephyr/cmake/modules/FindZephyr-sdk.cmake:61 (find_package):
  Could not find a configuration file for package "Zephyr-sdk" that is
  compatible with requested version "0.15".

  The following configuration files were considered but not accepted:

    /opt/zephyr-sdk-0.14.2/cmake/Zephyr-sdkConfig.cmake, version: 0.14.2
```

Reply:
actually the code flow is not changed with this PR.
And when using CMake package, we rely on the CMake message printing, not this code snippet.

This is code which originates all the way back to when Zephyr SDK 0.11.x was still supported:
https://github.com/zephyrproject-rtos/zephyr/blob/7f970093b33cf794352324ce820d82ea575bec0c/cmake/toolchain/zephyr/host-tools.cmake#L76

I believe this code can actually be removed now, but will examine closer to make sure there are no pifalls.

Thanks for spotting.
When using the compiler from Zephyr SDK, that is: `ZEPHYR_TOOLCHAIN_VARIANT==zephyr|<none>` then the standard messages from `find_package()` will be printed.

But if you are combining a 3rd party toolchain with host tools from Zephyr SDK, then this code gets activated, for example:

Reply:
code block removed.

Also with this change https://github.com/zephyrproject-rtos/zephyr/commit/07e4d085544c53ebd5d2c44101d40e0426389db1 this code block can no longer be reached.